topo_processor/util/aws_credentials.py

```diff
@@ -3,11 +3,32 @@
 import boto3
 
 credentials = None
+role_dict = {}
 
 
-def get_credentials():
+def get_credentials(bucket_name: str):
     global credentials
-    if not credentials:
-        session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
-        credentials = session.get_credentials()
+    if credentials:
+        return credentials
+
+    session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
```

Comment:
should move this outside of the function, we do not need a new session for every invocation